base.cmake

```diff
@@ -161,13 +161,15 @@ SET(CMAKE_PROJECT_${PROJECT_NAME}_INCLUDE "${CMAKE_CURRENT_LIST_DIR}/post-projec
 
 # Set a hook to finalize the setup, CMake will set CMAKE_CURRENT_LIST_DIR to "" at the end
 # Based off https://stackoverflow.com/questions/15760580/execute-command-or-macro-in-cmake-as-the-last-step-before-the-configure-step-f
-VARIABLE_WATCH(CMAKE_CURRENT_SOURCE_DIR SETUP_PROJECT_FINALIZE_HOOK)
+VARIABLE_WATCH(CMAKE_CURRENT_LIST_DIR SETUP_PROJECT_FINALIZE_HOOK)
 FUNCTION(SETUP_PROJECT_FINALIZE_HOOK VARIABLE ACCESS)
   IF("${${VARIABLE}}" STREQUAL "")
+    SET(CMAKE_CURRENT_LIST_DIR ${PROJECT_JRL_CMAKE_MODULE_DIR})
     SETUP_PROJECT_FINALIZE()
     IF(PROJECT_USE_CMAKE_EXPORT)
       SETUP_PROJECT_PACKAGE_FINALIZE()
     ENDIF()
+    SET(CMAKE_CURRENT_LIST_DIR "") # restore value
```

Comment:
Not sure to understand: 
- why the two `SET` are in the loop,
- and the *restore value*, which does not seem to *restore* anything.

Reply:
SETUP_PROJECT_FINALIZE_HOOK is called when CMAKE_CURRENT_LIST_DIR is changed, i.e. when it goes to "".
I use the restored value of CMAKE_CURRENT_LIST_DIR (${PROJECT_JRL_CMAKE_MODULE_DIR}), in order to let SETUP_PROJECT_FINALIZE and SETUP_PROJECT_PACKAGE_FINALIZE to get access to the right value, and not the empty string.

Reply:
Thanks for the explanation. May it is worth adding this in a comment as it is not straightforward.